compiler/rustc_resolve/src/imports.rs

```diff
@@ -284,39 +284,69 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         self.arenas.alloc_name_binding(NameBindingData {
             kind: NameBindingKind::Import { binding, import, used: Cell::new(false) },
             ambiguity: None,
+            warn_ambiguity: false,
             span: import.span,
             vis,
             expansion: import.parent_scope.expansion,
         })
     }
 
     /// Define the name or return the existing binding if there is a collision.
+    /// `update` indicates if the definition is a redefinition of an existing binding.
     pub(crate) fn try_define(
         &mut self,
         module: Module<'a>,
         key: BindingKey,
         binding: NameBinding<'a>,
+        update: bool,
```

Comment:
```suggestion
        warn_ambiguity: bool,
```
This flag doesn't mean an update anymore (in that case it would always be equal to `old_import == import` and implied `old_binding.is_some()`).
Now it's a flag that is used for downgrading errors into warning, and meaningless otherwise (or has a very convoluted meaning).

Reply:
I will both change `update` or `is_update` to `warn_ambiguity`